packages/headless/src/controllers/search-box/headless-search-box.ts

```diff
@@ -140,6 +141,32 @@ export function buildSearchBox(
       performSearch(logSearchboxSubmit());
     },
 
+    /**
+     * Add delimiters to a highlighted suggestion.
+     *
+     * @param highlightedSuggestion The highlighted suggestion.
+     * @param openingMatchDelimiter The opening delimiter for a match in the suggestion string (e.g. '<strong>')
+     * @param closingMatchDelimiter The closing delimiter for a match in the suggestion string (e.g. '</strong>')
+     * @param openingCorrectedDelimiter The opening delimiter for a correction in the suggestion string (e.g. '<i>')
+     * @param closingCorrectedDelimiter The closing delimiter for a correction in the suggestion string (e.g. '</i>')
+     */
+
+    formatHighlightedSuggestion(
+      highlightedSuggestion: string,
+      openingMatchDelimiter: string,
+      closingMatchDelimiter: string,
+      openingCorrectedDelimiter: string,
+      closingCorrectedDelimiter: string
```

Comment:
We should use an object add structure the parameters in a hierarchal way, allowing the user to specify on what they need. I would suggest the following:

```
interface SuggestionHighlightingOptions {
    exactMatchDelimiters?: {
        open: string;
        close: string;
    };
    correctionDelimiters?: {
        open: string;
        close: string;
    }
}
```
With this structure, the user has the freedom to specify only the highlights they are interested in (exactMatch or correction or both). When they specify, they need to pass both an opening and closing delimiter.

The `suggestion` is not passed because we have access to it in the state. We would return an array with all suggestions highlighted.